java/src/jmri/NamedBeanPropertyDescriptor.java

```diff
@@ -51,4 +53,22 @@ protected NamedBeanPropertyDescriptor(String propertyKey, E defaultValue) {
         return defaultValue.getClass();
     }
     
+    /**
+     * Equals based on Property Key and Default value Class.
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(Object obj) {
+        return (obj instanceof NamedBeanPropertyDescriptor && obj.hashCode() == this.hashCode());
+    }
+
+    /**
+     * hashCode based on Property Key and Default value Class.
+     * {@inheritDoc}
+     */
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(this.propertyKey) * Objects.hashCode(getValueClass());
```

Comment:
Do we know that defaultValue is always non-null?  `getValueClass` depends on that.

Reply:
Thanks. I cannot find a case where `defaultValue` is null and #9388 will ensure that it stays that way.